lib/ipmi_user.c

```diff
@@ -550,18 +588,22 @@ ipmi_user_test(struct ipmi_intf *intf, int argc, char **argv)
 	}
 	if (argc == 3) {
 		/* We need to prompt for a password */
-		password = ask_password(user_id);
-		if (!password) {
-			lprintf(LOG_ERR, "ipmitool: malloc failure");
+		memset(passbuf, 0, sizeof(passbuf));
+		if (ask_password(user_id, passbuf, sizeof(passbuf))) {
+			lprintf(LOG_ERR, "ipmitool: password failure");
+			memset(passbuf, 0, sizeof(passbuf));
 			return (-1);
 		}
+		password = passbuf;
 	} else {
 		password = argv[3];
 	}
-	return ipmi_user_test_password(intf,
+	ret = ipmi_user_test_password(intf,
 					 user_id,
 					 password,
 					 password_length == 20);
```

Comment:
Magic number - 20

Reply:
I'll make a define for the IPMI 1.5 vs 2.0 pw. ( I think it was 1.5 at least.).

Wasn't sure the tolerance on churn, but easy fix.